File: gin/src/Book/Application/CreateBookCommandHandler.go
package Application

type CreateBookCommandHandler struct {
	bookCreator BookCreator
}

func NewCreateBookCommandHandler(bookCreator BookCreator) *CreateBookCommandHandler {
	return &CreateBookCommandHandler{bookCreator: bookCreator}
}

func (cbh CreateBookCommandHandler) Handle(command CreateBookCommand) error {
	var title = command.Title
	var description = command.Description
	return cbh.bookCreator.Run(title, description)
}

File: express-typescript/src/Book/Application/CreateBookCommandHandler.ts
import {inject, injectable} from "tsyringe";
import {BookCreator} from "./BookCreator";
import {CreateBookCommand} from "./CreateBookCommand";
import {CommandHandler} from "../../Shared/Domain/CommandBus/CommandHandler";
import {Command} from "../../Shared/Domain/CommandBus/Command";

@injectable()
export class CreateBookCommandHandler implements CommandHandler<CreateBookCommand>{
    constructor(@inject("BookCreator") private bookCreator: BookCreator)
    {
    }

    async handle(command: CreateBookCommand): Promise<void>
    {
        const id = command.id;
        const title = command.title;
        const description = command.description;
        await this.bookCreator.run({id, title, description});
    }

    subscribedTo(): Command {
        return CreateBookCommand;
    }
}